### PR TITLE
Add support for "to" field in ChangeStreamDocument

### PR DESCRIPTION
**Pull request by jyemin**
https://jira.mongodb.org/browse/JAVA-3261

https://evergreen.mongodb.com/version/5ce59226e3c33108d18af9cb

### CODE REVIEW

File: driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java [hunk 1]
```diff
@@ -107,16 +109,43 @@ public ChangeStreamDocument(final BsonDocument resumeToken,
...
+     * @param updateDescription the update description
+     *
+     * @since 3.11
+     */
+    @BsonCreator
+    public ChangeStreamDocument(@BsonProperty("resumeToken") final BsonDocument resumeToken,
+                                @BsonProperty("ns") final BsonDocument namespaceDocument,
+                                @Nullable @BsonProperty("to") final BsonDocument destinationNamespaceDocument,
+                                @BsonProperty("fullDocument") final TDocument fullDocument,
+                                @Nullable @BsonProperty("documentKey") final BsonDocument documentKey,
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java]: Turns out more of these are nullable

File: driver-core/src/test/unit/com/mongodb/client/model/changestream/ChangeStreamDocumentCodecSpecification.groovy [hunk 2]
```diff
@@ -56,30 +57,37 @@ class ChangeStreamDocumentCodecSpecification extends Specification {
...
+        clazz << [Document, Document, Document]
         json << [
             '''{_id: {token: true}, ns: {db: "databaseName", coll: "collectionName"}, documentKey : {_id : 1},
                 fullDocument: {key: "value for fullDocument"}, clusterTime: { "$timestamp" : { "t" : 1234, "i" : 2 } }
                 operationType: "insert"}''',
             '''{_id: {token: true}, ns: {db: "databaseName", coll: "collectionName"}, documentKey : {_id : 2},
                 fullDocument: {key: "value for fullDocument"},
                 operationType: "update", updateDescription: {removedFields: ["a", "b"], updatedFields: {c: 1}}}''',
+            '''{_id: {token: true}, ns: {db: "databaseName", coll: "collectionName"},
+                to: {db: "databaseName2", coll: "collectionName2"}, operationType: "rename"}''',
```
Comment (rozza) [inline driver-core/src/test/unit/com/mongodb/client/model/changestream/ChangeStreamDocumentCodecSpecification.groovy]: The docs has a full event document, can we add a test to test we can parse all of that:

```
{
   _id : { <BSON Object> },
   "operationType" : "<operation>",
   "fullDocument" : { <document> },
   "ns" : {
      "db" : "<database>",
      "coll" : "<collection"
   },
   "to" : {
      "db" : "<database>",
      "coll" : "<collection"
   },
   "documentKey" : { "_id" : <value> },
   "updateDescription" : {
      "updatedFields" : { <document> },
      "removedFields" : [ "<field>", ... ]
   }
   "clusterTime" : <Timestamp>,
   "txnNumber" : <NumberLong>,
   "lsid" : {
      "id" : <UUID>,
      "uid" : <BinData>
   }
}
```
https://docs.mongodb.com/manual/reference/change-events/#change-stream-output

Alternatively, the docs gives examples of each event type..

File: driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java (hunk 1 above)
Comment (rozza) [reply driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java]: Should all of them to be nullable apart from resumeToken?

Comment (jyemin) [reply driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java]: `fullDocument` should.  I think the rest are properly non-nullable.

Comment (rozza) [reply driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java]: Depends what the user does with the pipeline?

Comment (jyemin) [reply driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java]: I'm comfortable with ChangeStreamDocument representing a pipeline whose structure hasn't been modified.  If they add fields, for example, ChangeStreamDocument is also inappropriate and they should decode to something else.

But open to other opinions on this too.

Comment (jyemin) [reply driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java]: I made everything null except resumeToken and operationType

File: driver-core/src/test/unit/com/mongodb/client/model/changestream/ChangeStreamDocumentCodecSpecification.groovy (hunk 2 above)
Comment (jyemin) [reply driver-core/src/test/unit/com/mongodb/client/model/changestream/ChangeStreamDocumentCodecSpecification.groovy]: Great idea. Helped me find all the nullable things in an unmodified change stream document.   

Done.

**Pull request closed without merging**